woocommerce/woocommerce-android
# [Coupons] Edit expiry date

Closes: #6494 


### Description
This PR adds support for editing the expiry date.
For the date picker, I had two options:
1. Use the MaterialDatePicker from the Fragment using events.
2. Create a Compsable date picker.

I went with the second approach, because it was a nice challenge, and also because Google is working on composable [implementations](https://issuetracker.google.com/issues/197942880), which means migrating to them would be easier.

Another point, regarding the scenario for deleting the expiry date, I went with a Dialog that shows two buttons for selecting the action (edit vs delete), this aligns with the iOS implementation (they use an action sheet).

### Testing instructions
1. Make sure coupons beta switch is enabled.
2. Open details of a coupon in the app, then navigate to edit using the menu.
3. Click on the expiry date.
4. If the date wasn't set, confirm that a picker is displayed.
5. If the date was already set, confirm that a dialog is displayed with two buttons.
6. Change the date, and confirm the changes are reflected.
7. Delete the date, and confirm the picker changes to "None"

### Images/gif
https://user-images.githubusercontent.com/1657201/168159961-6e15fcc6-4303-49f8-8ee6-edfae5f93e02.mov
